config-processor.js

```diff
@@ -0,0 +1,156 @@
+  } else if (typeof repoPlugins === 'object') {
+    // It is a merge manifest, handle excludes first
+    if (repoPlugins.exclude) {
+      plugins = applyExcludeList(orgPlugins, repoPlugins.exclude);
+    }
+
+    // Now handle includes
+    if (repoPlugins.include) {
+      plugins = applyIncludeList(pluginManager, plugins, repoPlugins.include, onInvalidPlugin);
+    }
+  }
+
+  config.plugins = plugins;
+
+  return config;
+};
+
+/**
+ * Apply the include list of plugin names and return the merged plugin list
+ *
+ * @param {PluginManager} pluginManager The plugin manager
+ * @param {PluginList} orgPlugins The list of plugins configured on the org
+ * @param {PluginList} repoIncludeList A list of plugins to merge into the config
+ * @param {OnInvalidPluginCallback} [onInvalidPlugin] Function to call when an invalid plugin is encountered
+ * @returns {PluginList} The filtered list of plugins
+ */
+// eslint-disable-next-line max-statements, complexity
+function applyIncludeList(pluginManager, orgPlugins, repoIncludeList, onInvalidPlugin) {
+  const plugins = deepClone(orgPlugins);
+  for (const pluginToInclude of repoIncludeList) {
+    const pluginName = typeof pluginToInclude === 'string' ? pluginToInclude : pluginToInclude.plugin;
+    const pluginInstance = pluginManager[pluginName];
+    if (!pluginInstance) {
+      onInvalidPlugin && onInvalidPlugin(pluginName);
+      // eslint-disable-next-line no-continue
+      continue;
+    }
+
+    if (typeof pluginToInclude === 'string' &&
+      !plugins.find(p => p === pluginToInclude ||
+        (/** @type {Plugin} */(p).plugin && /** @type {Plugin} */(p).plugin === pluginToInclude))) {
+      // If the entry in the include list is just a string, we can simply add it if it's not present.
+      plugins.push(pluginToInclude);
+    } else if (typeof pluginToInclude === 'object' && pluginToInclude.plugin) {
+      // If the entry in the include list is an object, we have to be more careful.
+      const existingPlugin = plugins.find(
+        p => p === pluginToInclude.plugin ||
+        (/** @type {Plugin} */(p).plugin && /** @type {Plugin} */(p).plugin === pluginToInclude.plugin));
+      if (!existingPlugin) {
+        // If it's not in the existing list, just add it directly
+        plugins.push(pluginToInclude);
+      } else if (typeof existingPlugin === 'string' || !existingPlugin.config) {
+        // If it is in the existing list but only as a string, we can just replace the string with the object.
+        // Same story if the existing plugin has no config field set.
+        const idx = plugins.indexOf(existingPlugin);
+        plugins.splice(idx, 1, pluginToInclude);
+      } else {
+        // If it is in the existing list as an object, and the plugin to include has config specified, we have to merge
+        // the config
+        existingPlugin.config = pluginInstance.mergeConfig(existingPlugin.config, pluginToInclude.config);
+      }
+    }
+  }
```

Comment:
Have not tested this, but I believe this makes the control flow a bit easier to read by not bi-furcating based on the `typeof` for `pluginToInclude`:

``` js
const pluginEqual = {
  literal(x, y) {
    if (typeof x !== 'string') return false;
    return x === y
      || (/** @type {Plugin} */(y).plugin && /** @type {Plugin} */(y).plugin === x);
  },
  
  obj(x, y) {
    if (typeof x !== 'object') return false;
    return x.plugin === y 
      || (/** @type {Plugin} */(y).plugin && /** @type {Plugin} */(y).plugin === x.plugin);
  }
};

const plugins = deepClone(orgPlugins);
for (const pluginToInclude of repoIncludeList) {
  const pluginName = typeof pluginToInclude === 'string' ? pluginToInclude : pluginToInclude.plugin;
  const pluginInstance = pluginManager[pluginName];
  if (!pluginInstance) {
    onInvalidPlugin && onInvalidPlugin(pluginName);
    // eslint-disable-next-line no-continue
    continue;
  }

  const existingPlugin = plugins.find(p => {
    return pluginEqual.literal(pluginToInclude, p)
      || pluginEqual.obj(pluginToInclude, p)
  });

  if (!existingPlugin) {
    // If it's not in the existing list, just add it directly
    plugins.push(pluginToInclude);
    continue; 
  } 

  if (typeof pluginToInclude === 'object') {
    if (typeof existingPlugin === 'string' || !existingPlugin.config) {
      // If it is in the existing list but only as a string, we can just replace the string with the object.
      // Same story if the existing plugin has no config field set.
      const idx = plugins.indexOf(existingPlugin);
      plugins.splice(idx, 1, pluginToInclude);
    } else {
      // If it is in the existing list as an object, and the plugin to include has config specified, we have to merge
      // the config
      existingPlugin.config = pluginInstance.mergeConfig(existingPlugin.config, pluginToInclude.config);
    }
  }
}
```